### ISSUE TITLE
Not working with Neo4j 3.1.0

### ISSUE THREAD

**Issue by nadworny** [graphaware/neo4j-resttest#3]
I'm trying to setup graphaware-resttest-3.1.0.44.15.jar with Neo4j 3.1.0 CE but without success. I've tried different 3.1.* neo4j version but always getting the same error. I've tried also 3.0.7 together with graphaware-resttest-3.0.7.44.15.jar but also no success... Would appreciate your help!

If I start neo4j only with the framework plugin, everything works ok. If both jar files are inside, the error occurs. I have updated the neo4j.conf with `dbms.unmanaged_extension_classes=com.graphaware.server=/graphaware`

I'm on MacOS using docker images but the same happens when I use the dmg installer.

```
Starting Neo4j.
2017-03-30 11:49:33.533+0000 INFO  Starting...
2017-03-30 11:49:34.217+0000 INFO  Bolt enabled on 0.0.0.0:7687.
2017-03-30 11:49:36.511+0000 INFO  [c.g.r.b.RuntimeKernelExtension] GraphAware Runtime disabled.
2017-03-30 11:49:47.735+0000 ERROR Failed to start Neo4j: Starting Neo4j failed: Component 'org.neo4j.server.database.LifecycleManagingDatabase@5058a8e2' was successfully initialized, but failed to start. Please see attached cause exception. Starting Neo4j failed: Component 'org.neo4j.server.database.LifecycleManagingDatabase@5058a8e2' was successfully initialized, but failed to start. Please see attached cause exception.
org.neo4j.server.ServerStartupException: Starting Neo4j failed: Component 'org.neo4j.server.database.LifecycleManagingDatabase@5058a8e2' was successfully initialized, but failed to start. Please see attached cause exception.
	at org.neo4j.server.exception.ServerStartupErrors.translateToServerStartupError(ServerStartupErrors.java:68)
	at org.neo4j.server.AbstractNeoServer.start(AbstractNeoServer.java:227)
	at org.neo4j.server.ServerBootstrapper.start(ServerBootstrapper.java:91)
	at org.neo4j.server.ServerBootstrapper.start(ServerBootstrapper.java:68)
	at org.neo4j.server.CommunityEntryPoint.main(CommunityEntryPoint.java:28)
Caused by: org.neo4j.kernel.lifecycle.LifecycleException: Component 'org.neo4j.server.database.LifecycleManagingDatabase@5058a8e2' was successfully initialized, but failed to start. Please see attached cause exception.
	at org.neo4j.kernel.lifecycle.LifeSupport$LifecycleInstance.start(LifeSupport.java:443)
	at org.neo4j.kernel.lifecycle.LifeSupport.start(LifeSupport.java:107)
	at org.neo4j.server.AbstractNeoServer.start(AbstractNeoServer.java:199)
	... 3 more
Caused by: java.lang.RuntimeException: Error starting org.neo4j.kernel.impl.factory.GraphDatabaseFacadeFactory, /var/lib/neo4j/data/databases/graph.db
	at org.neo4j.kernel.impl.factory.GraphDatabaseFacadeFactory.initFacade(GraphDatabaseFacadeFactory.java:199)
	at org.neo4j.kernel.impl.factory.GraphDatabaseFacadeFactory.newFacade(GraphDatabaseFacadeFactory.java:130)
	at org.neo4j.server.CommunityNeoServer.lambda$static$0(CommunityNeoServer.java:57)
	at org.neo4j.server.database.LifecycleManagingDatabase.start(LifecycleManagingDatabase.java:89)
	at org.neo4j.kernel.lifecycle.LifeSupport$LifecycleInstance.start(LifeSupport.java:433)
	... 5 more
Caused by: org.neo4j.kernel.lifecycle.LifecycleException: Component 'org.neo4j.kernel.impl.proc.Procedures@35752757' was successfully initialized, but failed to start. Please see attached cause exception.
	at org.neo4j.kernel.lifecycle.LifeSupport$LifecycleInstance.start(LifeSupport.java:443)
	at org.neo4j.kernel.lifecycle.LifeSupport.start(LifeSupport.java:107)
	at org.neo4j.kernel.impl.factory.GraphDatabaseFacadeFactory.initFacade(GraphDatabaseFacadeFactory.java:195)
	... 9 more
Caused by: org.neo4j.kernel.api.exceptions.ProcedureException: Conflicting procedure annotation, cannot use PerformsWrites and mode
	at org.neo4j.kernel.impl.proc.ReflectiveProcedureCompiler.compileProcedure(ReflectiveProcedureCompiler.java:168)
	at org.neo4j.kernel.impl.proc.ReflectiveProcedureCompiler.compileProcedure(ReflectiveProcedureCompiler.java:133)
	at org.neo4j.kernel.impl.proc.ProcedureJarLoader.loadProcedures(ProcedureJarLoader.java:92)
	at org.neo4j.kernel.impl.proc.ProcedureJarLoader.loadProceduresFromDir(ProcedureJarLoader.java:80)
	at org.neo4j.kernel.impl.proc.Procedures.start(Procedures.java:208)
	at org.neo4j.kernel.lifecycle.LifeSupport$LifecycleInstance.start(LifeSupport.java:433)
	... 11 more
```

**Comment by nadworny** [comment]
I've just build this project with mvn install and the deployment worked fine but now I'm getting the following error while executing assertSameGraph POST request:

```
Starting Neo4j.
2017-03-30 14:39:20.860+0000 INFO  Starting...
2017-03-30 14:39:21.544+0000 INFO  Bolt enabled on 0.0.0.0:7687.
2017-03-30 14:39:23.092+0000 INFO  [c.g.r.b.RuntimeKernelExtension] GraphAware Runtime disabled.
2017-03-30 14:39:30.518+0000 INFO  Started.
2017-03-30 14:39:30.766+0000 INFO  [c.g.s.f.b.GraphAwareServerBootstrapper] started
2017-03-30 14:39:30.770+0000 INFO  Mounted unmanaged extension [com.graphaware.server] at [/graphaware]
2017-03-30 14:39:30.998+0000 INFO  Mounting GraphAware Framework at /graphaware
2017-03-30 14:39:31.023+0000 INFO  Will try to scan the following packages: {com.**.graphaware.**,org.**.graphaware.**,net.**.graphaware.**}
2017-03-30 14:39:32.909+0000 INFO  Remote interface available at http://localhost:7474/
2017-03-30 14:42:24.163+0000 WARN   Handler processing failed; nested exception is java.lang.NoClassDefFoundError: com/graphaware/test/unit/GraphUnit
org.springframework.web.util.NestedServletException: Handler processing failed; nested exception is java.lang.NoClassDefFoundError: com/graphaware/test/unit/GraphUnit
	at org.springframework.web.servlet.DispatcherServlet.triggerAfterCompletionWithError(DispatcherServlet.java:1303)
	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:977)
	at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:893)
	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:970)
	at org.springframework.web.servlet.FrameworkServlet.doPost(FrameworkServlet.java:872)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:707)
	at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:846)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:808)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1669)
	at org.neo4j.server.rest.dbms.AuthorizationEnabledFilter.doFilter(AuthorizationEnabledFilter.java:122)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652)
	at com.graphaware.server.tx.LongRunningTransactionFilter.doFilter(LongRunningTransactionFilter.java:88)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652)
	at com.graphaware.server.foundation.bootstrap.GraphAwareBootstrappingFilter.doFilter(GraphAwareBootstrappingFilter.java:85)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652)
	at org.neo4j.server.rest.web.CollectUserAgentFilter.doFilter(CollectUserAgentFilter.java:69)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:585)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:221)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1127)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:515)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:185)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1061)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
	at org.eclipse.jetty.server.handler.HandlerList.handle(HandlerList.java:52)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:97)
	at org.eclipse.jetty.server.Server.handle(Server.java:497)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:310)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:257)
	at org.eclipse.jetty.io.AbstractConnection$2.run(AbstractConnection.java:540)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:635)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:555)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.NoClassDefFoundError: com/graphaware/test/unit/GraphUnit
	at com.graphaware.module.resttest.RestTestApi.clearDatabase(RestTestApi.java:55)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:221)
	at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:137)
	at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:110)
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:806)
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:729)
	at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:85)
	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:959)
	... 32 more
Caused by: java.lang.ClassNotFoundException: com.graphaware.test.unit.GraphUnit
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 44 more
```

**Comment by nadworny** [comment]
Added the following line to the neo4j.conf:
`com.graphaware.runtime.enabled=true`

Now the runtime is starting properly but still getting an errror:
```
Starting Neo4j.
2017-03-30 14:55:20.899+0000 INFO  Starting...
2017-03-30 14:55:21.598+0000 INFO  Bolt enabled on 0.0.0.0:7687.
2017-03-30 14:55:23.116+0000 INFO  [c.g.r.b.RuntimeKernelExtension] GraphAware Runtime enabled, bootstrapping...
2017-03-30 14:55:23.135+0000 INFO  Running in a single-node architecture (Neo4j Community)
2017-03-30 14:55:23.149+0000 INFO  [c.g.r.b.RuntimeKernelExtension] GraphAware Runtime bootstrapped, starting the Runtime...
2017-03-30 14:55:28.238+0000 INFO  Starting GraphAware...
2017-03-30 14:55:28.240+0000 INFO  Loading module metadata...
2017-03-30 14:55:28.305+0000 INFO  Module metadata loaded.
2017-03-30 14:55:28.307+0000 INFO  Starting transaction-driven modules...
2017-03-30 14:55:28.308+0000 INFO  Transaction-driven modules started.
2017-03-30 14:55:28.310+0000 INFO  There are no timer-driven runtime modules. Not scheduling any tasks.
2017-03-30 14:55:28.311+0000 INFO  GraphAware started.
2017-03-30 14:55:28.314+0000 INFO  [c.g.r.b.RuntimeKernelExtension] GraphAware Runtime automatically started.
2017-03-30 14:55:30.218+0000 INFO  Started.
2017-03-30 14:55:30.455+0000 INFO  [c.g.s.f.b.GraphAwareServerBootstrapper] started
2017-03-30 14:55:30.457+0000 INFO  Mounted unmanaged extension [com.graphaware.server] at [/graphaware]
2017-03-30 14:55:30.669+0000 INFO  Mounting GraphAware Framework at /graphaware
2017-03-30 14:55:30.691+0000 INFO  Will try to scan the following packages: {com.**.graphaware.**,org.**.graphaware.**,net.**.graphaware.**}
2017-03-30 14:55:32.423+0000 INFO  Remote interface available at http://localhost:7474/
2017-03-30 14:56:00.371+0000 WARN  Failed to read HTTP message: org.springframework.http.converter.HttpMessageNotReadableException: Required request body is missing: public java.lang.String com.graphaware.module.resttest.RestTestApi.assertSameGraph(com.graphaware.module.resttest.RestTestRequest,javax.servlet.http.HttpServletResponse) throws java.io.IOException
2017-03-30 14:56:00.375+0000 WARN  Handler execution resulted in exception: Required request body is missing: public java.lang.String com.graphaware.module.resttest.RestTestApi.assertSameGraph(com.graphaware.module.resttest.RestTestRequest,javax.servlet.http.HttpServletResponse) throws java.io.IOException
2017-03-30 14:56:41.663+0000 WARN   Handler processing failed; nested exception is java.lang.NoClassDefFoundError: com/graphaware/test/unit/GraphUnit
org.springframework.web.util.NestedServletException: Handler processing failed; nested exception is java.lang.NoClassDefFoundError: com/graphaware/test/unit/GraphUnit
	at org.springframework.web.servlet.DispatcherServlet.triggerAfterCompletionWithError(DispatcherServlet.java:1303)
	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:977)
	at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:893)
	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:970)
	at org.springframework.web.servlet.FrameworkServlet.doPost(FrameworkServlet.java:872)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:707)
	at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:846)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:808)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1669)
	at org.neo4j.server.rest.dbms.AuthorizationEnabledFilter.doFilter(AuthorizationEnabledFilter.java:122)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652)
	at com.graphaware.server.tx.LongRunningTransactionFilter.doFilter(LongRunningTransactionFilter.java:88)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652)
	at com.graphaware.server.foundation.bootstrap.GraphAwareBootstrappingFilter.doFilter(GraphAwareBootstrappingFilter.java:85)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652)
	at org.neo4j.server.rest.web.CollectUserAgentFilter.doFilter(CollectUserAgentFilter.java:69)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:585)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:221)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1127)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:515)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:185)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1061)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
	at org.eclipse.jetty.server.handler.HandlerList.handle(HandlerList.java:52)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:97)
	at org.eclipse.jetty.server.Server.handle(Server.java:497)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:310)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:257)
	at org.eclipse.jetty.io.AbstractConnection$2.run(AbstractConnection.java:540)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:635)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:555)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.NoClassDefFoundError: com/graphaware/test/unit/GraphUnit
	at com.graphaware.module.resttest.RestTestApi.assertSameGraph(RestTestApi.java:68)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:221)
	at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:137)
	at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:110)
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:806)
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:729)
	at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:85)
	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:959)
	... 32 more
Caused by: java.lang.ClassNotFoundException: com.graphaware.test.unit.GraphUnit
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 44 more
```

**Comment by szenyo** [comment]
Itt will be fixed in the next release.

**Comment by szenyo** [comment]
It was fixed in graphaware-resttest-3.1.4.49.16.jar
